### PR TITLE
solveset(domain=S.Reals) rewrites the function in the answer. #10425

### PR DESCRIPTION
**Pull request by shubhamtibra**
If the expression is not solved then rewrite it in the form of trigonometric functions.

### CODE REVIEW

**Comment by smichr** [comment]
Make sure to add tests for each case to show what this is doing. Probably in test_solveset.py in the function that tests trigonometric functions.

**Comment by shubhamtibra** [comment]
The output for this `solveset(sin(x+y)-sin(x),y,domain=S.Reals)` after changing the files is 
`ConditionSet(y, Eq(-sin(x) + sin(x + y), 0), (-oo, oo))` . But when I am adding  
`assert solveset(sin(x + a) - sin(x), a, domain=S.Reals) == \
        ConditionSet(a, Eq(-sin(x) + sin(x + a), 0), S.Reals)`
to the tests it is giving assertion error. I can't figure out why this is happening

**Comment by smichr** [comment]
It would be better to simply capture the original function at the start and send back the solution in terms of that:

```python
f_orig = f
...
else:
    return ConditionSet(symbol, Eq(f_orig, 0), S.Reals)
```

**Comment by shubhamtibra** [comment]
Oh, didn't think of that. Doing it now.

**Comment by shubhamtibra** [comment]
Ping @smichr @asmeurer

File: sympy/solvers/solveset.py [hunk 1]
```diff
@@ -538,6 +538,7 @@ def _solve_as_rational(f, symbol, solveset_solver, as_poly_solver):
 def _solve_real_trig(f, symbol):
     """ Helper to solve trigonometric equations """
     f = trigsimp(f)
+    f_original = f
```
Comment (smichr) [inline sympy/solvers/solveset.py]: It's no longer the original function at this point. I would move that line to before the trigsimp line.

**Comment by shubhamtibra** [comment]
@smichr I tried that, but this following test failed 
`assert solveset(Eq(sin(x)**2 + cos(x)**2, 1), x, domain=S.Reals) == \
        ConditionSet(x, True, S.Reals)`

**Comment by shubhamtibra** [comment]
Also I think if the expression can be simplified then it is better to return that. For instance
`solveset(sin(x)**2 + cos(x)**2 + sin(x + y) - sin(x), y, domain=S.Reals)` returning  
`ConditionSet(y, Eq(-sin(x) + sin(x + y) + 1, 0), (-oo, oo))` is cooler than returning the original expression.

**Comment by shubhamtibra** [comment]
@smichr Can you review it please?

**Comment by smichr** [comment]
looks good and I agree about the simpler answer being better

**Comment by shubhamtibra** [comment]
Thanks!

**Pull request merged**